### PR TITLE
Use a swap in place of a copy.

### PR DESCRIPTION
**Pull request by drwells**
A small optimization: I found this when working on something else.

### CODE REVIEW

File: source/base/partitioner.cc [hunk 1]
```diff
@@ -277,7 +278,8 @@ namespace Utilities
               import_targets_temp.push_back(std::pair<unsigned int,
                                             unsigned int> (i, receive_buffer[i]));
             }
-        import_targets_data = import_targets_temp;
+
+        swap(import_targets_temp, import_targets_data);
```
Comment (bangerth) [inline source/base/partitioner.cc]: I think the cleaner approach would actually be to use
```
  import_targets_data = std::move(import_targets_temp);
```
or something similar. This makes it clear to the reader which variable survives.

Comment (drwells) [inline source/base/partitioner.cc]: I agree. If we do this, though, then we should wait until the C++11 support requirement is merged.

**Comment by drwells** [comment]
Now blocked by #4106.

**Comment by drwells** [comment]
We are officially incompatible with C++98 now :)

**Pull request merged**